typescript/cli/src/deploy/core.ts

```diff
@@ -1,461 +1,66 @@
-import { confirm } from '@inquirer/prompts';
-import { ethers } from 'ethers';
-
-import { ChainAddresses, IRegistry } from '@hyperlane-xyz/registry';
 import {
-  ChainMap,
   ChainName,
   CoreConfig,
-  GasOracleContractType,
-  HooksConfig,
-  HyperlaneAddressesMap,
-  HyperlaneContractsMap,
-  HyperlaneCore,
-  HyperlaneCoreDeployer,
-  HyperlaneIsmFactory,
-  HyperlaneProxyFactoryDeployer,
-  IgpConfig,
-  IsmConfig,
-  IsmType,
-  MultisigConfig,
-  RoutingIsmConfig,
-  buildAgentConfig,
-  buildAggregationIsmConfigs,
-  defaultMultisigConfigs,
-  multisigIsmVerificationCost,
-  serializeContractsMap,
+  DeployedCoreAdresses,
+  EvmCoreModule,
 } from '@hyperlane-xyz/sdk';
-import { Address, objFilter, objMap, objMerge } from '@hyperlane-xyz/utils';
 
-import { presetHookConfigs, readHooksConfigMap } from '../config/hooks.js';
-import { readIsmConfig } from '../config/ism.js';
-import { readMultisigConfig } from '../config/multisig.js';
 import { MINIMUM_CORE_DEPLOY_GAS } from '../consts.js';
 import { WriteCommandContext } from '../context/types.js';
-import {
-  log,
-  logBlue,
-  logBoldUnderlinedRed,
-  logGray,
-  logGreen,
-  logRed,
-} from '../logger.js';
-import { runMultiChainSelectionStep } from '../utils/chains.js';
-import { runFileSelectionStep, writeJson } from '../utils/files.js';
+import { readYamlOrJson } from '../utils/files.js';
 
 import {
   completeDeploy,
-  isISMConfig,
-  isZODISMConfig,
   prepareDeploy,
+  runDeployPlanStep,
   runPreflightChecksForChains,
 } from './utils.js';
 
-const CONTRACT_CACHE_EXCLUSIONS = ['interchainGasPaymaster'];
-
+interface DeployParams {
+  context: WriteCommandContext;
+  chain: ChainName;
+  config: CoreConfig;
+}
 /**
  * Executes the core deploy command.
  */
-export async function runCoreDeploy({
+export async function coreDeploy({
```

Comment:
Personally I like when function names begin with an action verb (e.g. `run`, `deploy`) though I'll sometimes make exceptions for pure functions that just transform input (`fooToBar`)

Reply:
fixed